docs/source/cpp/threading.rst

```diff
@@ -0,0 +1,100 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: cpp
+.. highlight:: cpp
+
+.. _cpp_thread_management:
+
+=================
+Thread Management
+=================
+
+.. seealso::
+   :doc:`Thread management API reference <api/thread>`
+
+Thread Pools
+=======================
+
+Many Arrow operations distribute work across multiple threads to take
+advantage of underlying hardware parallelism.  For example, when reading a
+parquet file we can decode each column in parallel.  To achieve this we
```

Comment:
Nit: "Parquet" capitalized, also let's cross-reference to the corresponding doc page?

Reply:
I'm not sure exactly where the best place would be to cross-reference.  I added a reference to https://arrow.apache.org/docs/cpp/api/formats.html#_CPPv4N7parquet5arrow10FileReader15set_use_threadsEb